packages/create-amplify/src/get_project_root.ts

```diff
@@ -16,6 +16,10 @@ export const getProjectRoot = async () => {
         defaultValue: defaultProjectRoot,
       });
 
+  if (projectRoot != defaultProjectRoot && projectRoot.startsWith('/')) {
+    projectRoot = projectRoot.substring(1);
+  }
```

Comment:
Paths prefixed by `/` are absolute paths on Linux/MacOS (see [here](https://www.linuxfoundation.org/blog/blog/classic-sysadmin-absolute-path-vs-relative-path-in-linux-unix)).
We should not be stripping `/` as it might be intentional input.

Reply:
Rather than strip off the leading /, we need to figure out why the [mkdir call](https://nodejs.org/api/module.html#modulecreaterequirefilename) is not working for absolute paths

Reply:
> Rather than strip off the leading /, we need to figure out why the [mkdir call](https://github.com/aws-amplify/amplify-backend/pull/857/files#diff-17a8da21bcc08aa19a0287db1113f604841ad4119d227505260fdc18ac081181R34) is not working for absolute paths

The original issue was about the case where `mkdir /abc` was failing, but mkdir fails on my Unix system also if I call it and ask it to create an absolute path that doesn't exist, ie. `mkdir /abc`. So is it really a bug or perhaps expected that it should fail?

Reply:
thank you message 🙇 @brandoncroberts !!

Today I looked for why it would be Error.
When call` fsp.mkdir`, input pathName maybe neccesary exist current the hierarchy or under a directory.

> https://github.com/aws-amplify/amplify-backend/pull/857#issuecomment-1878034992


#### shot test case.
```typescript
import path from 'path';
import fsp from 'fs/promises';

const userInputs = [
    "/test/test",
    path.resolve(process.cwd(), "/tests", "root"),
]

async function createDir(pathName: string) {
    const isAbsolute = path.isAbsolute(pathName)
    console.log(`current dir: ${process.cwd()}`)
    console.log(`user input dir: ${pathName}`)

    if (isAbsolute) {
        //NOTE: if the dir is above where fsp.mkdir is done, Fails
        try {
            //success "path.resolve(process.cwd(), "/tests", "root")"
            await fsp.mkdir(pathName, {recursive: true})
        } catch (e) {
            //Solution Success "/test/test"
            await fsp.mkdir(path.join(process.cwd(), pathName), {recursive: true})
        }
    }
}

userInputs.map(async (userInput, index) => {
    console.log("==========================")
    console.log(`test count: ${index}`)
    await createDir(userInput)
})
```

Reply:
> So is it really a bug or perhaps expected that it should fail?

Most likely it's expected to fail. I.e. I'd expect `mkdir /abc` to fail. However, I'm not sure if error mentioned in https://github.com/aws-amplify/amplify-backend/issues/827 is a good one (I'd expect it to fail with permission error instead).
Attempt too `mkdir -p /Users/<someusername>/foo/bar` should succeed on the other hand. (-p is equivalent of `recursive: true`)

Reply:
True that the error message could be better. On MacOS the error is `mkdir: /abc: Read-only file system`, which seems to be a MacOS specific Error on Catalina and higher versions, some discussion I found on that here.
https://stackoverflow.com/a/58404057
Maybe Linux has different permissions, could be tested.

And also true that  `mkdir /Users/some-user/some-folder/foobar` works to create a directory on MacOS.
MacOS just doesn't seem to allow you to create a directory at the `/` level.